orangewidget/settings.py

```diff
@@ -719,6 +714,79 @@ def is_allowed_type(cls, tp) -> bool:
             return args[0] in (str, bool, int, float) \
                    and cls.is_allowed_type(args[1])
 
+    @classmethod
+    def check_type(cls, value, tp) -> bool:
```

Comment:
How do you annotate `tp` here? It is a Union of Setting, type and ... the things you get from typing, like List or Union or Tuple ... The latter don't have a common, accessible base type, AFAIK.